subworkflows/local/yaml_input/main.nf

```
@@ -83,3 +84,61 @@ workflow YAML_INPUT {
+
+    // Initialise defaults
+    def fofn_files = []
+    def direct_files = []
+    def files_list = data_files instanceof List ? data_files : [data_files]
+
+    // Process each file - separate FOFN from direct files
+        files_list.each { file_path ->
+            if (file_path.toString().contains(".fofn")) {
+                def fofn_content = file(file_path).text.split('\n')
+                    .collect { it.trim() }
+                    .findAll { it } // Remove empty lines
+                fofn_files.addAll(fofn_content)
+            } else {
+                direct_files.add(file_path)
+            }
+        }
+
+        // Combine all files
+        def all_files = direct_files + fofn_files
+
+        // Validate files based on data type
+        if (data_type == "cram") {
+            def invalid_files = all_files.findAll {
+                !it.toString().contains(".cram")
+            }
+            if (invalid_files.size() > 0) {
+                error "One of the input hic files does not match cram format. Invalid files: ${invalid_files}"
+            }
+        } else if (data_type == "longread") {
+            def invalid_files = all_files.findAll {
+                !it.toString().contains(".fasta.gz") &&
+                !it.toString().contains(".fa.gz") &&
+                !it.toString().contains(".fn.gz")
+            }
+            if (invalid_files.size() > 0) {
+                error "One of the input longread files does not match expected formats (fn.gz, fa.gz, fasta.gz). Invalid files: ${invalid_files}"
+            }
+        }
+
+        // Create the resolved channel tuple
+        def resolved_channel = tuple(
+            [
+                id:         tolid_ver,
+                single_end: read_type != "illumina",
+                aligner:    read_type && data_type == "cram" ? read_type : "NA",
+                read_type:  read_type,
+            ],
+            all_files.collect { file(it, checkIfExists: true) }.unique()
```

I would probably raise an error if the inputs were not unique